runtime/lua/vim/_editor.lua

```diff
@@ -1112,6 +1112,26 @@ function vim._init_default_mappings()
     do_open(get_visual_selection())
   end, { desc = gx_desc })
 
+  -- z=
+  local spell_on_choice = vim.schedule_wrap(function(_, idx)
+    if type(idx) == 'number' then
+      vim.cmd.normal({ idx .. 'z=', bang = true })
+    end
+  end)
+  local spell_select = function()
+    if vim.v.count > 0 then
+      spell_on_choice(nil, vim.v.count)
+      return
+    end
+    local cword = vim.fn.expand('<cword>')
+    vim.ui.select(
+      vim.fn.spellsuggest(cword, vim.o.lines),
```

Comment:
I think `vim.o.lines` is at least off by one because of the `Type number and <Enter> or click with the mouse (q or empty cancels):` line.
`vim.o.lines - 1` seems better, but it's still wrong if the 'Type number..` line is wrapped on the screen (try making the window very narrow and then doing a `z=`).

Reply:
Yeah, having `vim.o.lines - 2` might be a better idea if the goal is to emulate default `z=`. But this number is pretty much arbitrary, so might as well be `math.floor(0.5 * vim.o..lines)`.

Reply:
Why not just remove the argument and use the default maximum of 25?

Reply:
That is more a design question, really. Built-in `z=` uses whole screen to populate spell suggestions, so I'd expect it to be that way. Also, if instance height is less than 25-27, then it might not play nicely with some `vim.ui.select` overrides.